sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3629,6 +3629,15 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val JSON_ENABLE_PARTIAL_RESULTS =
+    buildConf("spark.sql.json.enablePartialResults")
+      .internal()
+      .doc("When set to true, enables partial results for structs, maps, and arrays in JSON " +
+        "when one or more fields do not match the schema")
+      .version("3.4.0")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
I am still debating whether to keep this as `true` or `false`. On one hand, when enabled, it fixes the correctness issue of partially parsing JSON records. When disabled, it could cause performance issues (impact is yet to be confirmed).

@MaxGekk I would love to know your thoughts on this. I would prefer to keep it enabled until the benchmark is produced that shows the regression.

Reply:
Could you re-gen results of `JsonBenchmark`. If there are no significant diffs, I am ok to enable it. Also, it would be nice to add one more benchmark for this particular case of partial results.

Reply:
Yes, I will do that, thanks!

Reply:
I have rerun the benchmarks and posted results in the comments. I think I would rather keep it disabled until I find a way to make the code faster.

Reply:
@sadikovi Could you enable it by default since there is no regression.

Reply:
Yes, I will enable the flag by default.

Reply:
Done. I have enabled the flag by default.